core/cli/src/main/resources/bin/_module_opts.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# JDK 9+ modules we need
+JAVA_MODULES=(
+        java.base \
+        java.compiler \
+        java.datatransfer \
+        java.desktop \
+        java.instrument \
+        java.logging \
+        java.management \
+        java.management.rmi \
+        java.naming \
+        java.prefs \
+        java.rmi \
+        java.scripting \
+        java.security.jgss \
+        java.security.sasl \
+        java.sql \
+        java.sql.rowset \
+        java.xml \
+        jdk.attach \
+        jdk.httpserver \
+        jdk.jdi \
+        jdk.sctp \
+        jdk.security.auth \
+        jdk.xml.dom \
+)
+
+# JDK 9+ modules that need to read from public methods
+JAVA_READS=(
+    "java.xml=java.logging"
+)
+
+# JDK 9+ modules that need public methods available to reflection
+JAVA_EXPORTS=(
+    "java.base/sun.net.www.protocol.file=ALL-UNNAMED" \
+    "java.base/sun.net.www.protocol.ftp=ALL-UNNAMED" \
+    "java.base/sun.net.www.protocol.http=ALL-UNNAMED" \
+    "java.base/sun.net.www.protocol.https=ALL-UNNAMED" \
+    "java.base/sun.net.www.protocol.jar=ALL-UNNAMED" \
+    "java.base/sun.net.www.content.text=ALL-UNNAMED" \
+    "java.base/sun.security.x509=ALL-UNNAMED" \
+    "jdk.xml.dom/org.w3c.dom.html=ALL-UNNAMED" \
+    "jdk.naming.rmi/com.sun.jndi.url.rmi=ALL-UNNAMED" \
+    "java.rmi/sun.rmi.registry=ALL-UNNAMED" \
+    "java.security.sasl/com.sun.security.sasl=ALL-UNNAMED" \
+    "org.apache.karaf.specs/org.apache.karaf.specs.locator=java.xml,ALL-UNNAMED"
```

Comment:
This gets output on startup now: `WARNING: Unknown module: org.apache.karaf.specs specified to --add-exports`